syntax = "proto3";

package ai.verta.monitoring;
option go_package = "github.com/VertaAI/modeldb/protos/gen/go/protos/public/monitoring";

option java_multiple_files = true;

import "google/api/annotations.proto";
import "google/protobuf/struct.proto";

import "monitoring/Summary.proto";

// Request to fetch the existing labels for a given subset of summary samples
message FindSampleLabelsRequest {
    FilterQuerySummarySample filter = 1; // Optional

    message Response {
        repeated string labels = 1;
    }
}

// Request to fetch the existing label values for a given subset of summary samples
message FindSampleLabelValuesRequest {
    FilterQuerySummarySample filter = 1; // Optional
    repeated string labels = 2;

    message Response {
        map<string, FindSampleLabelValuesItem> labels = 1;
    }
}

message FindSampleLabelValuesItem {
    repeated string values = 1;
}

service LabelService {
    rpc findLabels(FindSampleLabelsRequest) returns (FindSampleLabelsRequest.Response) {
        option (google.api.http) = {
            post: "/v1/labels/findLabels"
            body: "*"
        };
    };

    rpc findLabelValues(FindSampleLabelValuesRequest) returns (FindSampleLabelValuesRequest.Response) {
        option (google.api.http) = {
            post: "/v1/labels/findLabelValues"
            body: "*"
        };
    };
}
